shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionReducer.kt

```diff
@@ -0,0 +1,101 @@
+package org.hyperskill.app.step_completion.presentation
+
+import org.hyperskill.app.step.domain.model.StepRoute
+import org.hyperskill.app.step_completion.domain.analytic.StepCompletionClickedContinueHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.domain.analytic.StepCompletionClickedStartPracticingHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.domain.analytic.topic_completed_modal.StepCompletionTopicCompletedModalClickedGoToHomeScreenHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.domain.analytic.topic_completed_modal.StepCompletionTopicCompletedModalHiddenHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.domain.analytic.topic_completed_modal.StepCompletionTopicCompletedModalShownHyperskillAnalyticEvent
+import org.hyperskill.app.step_completion.presentation.StepCompletionFeature.Action
+import org.hyperskill.app.step_completion.presentation.StepCompletionFeature.ContinueButtonAction
+import org.hyperskill.app.step_completion.presentation.StepCompletionFeature.Message
+import org.hyperskill.app.step_completion.presentation.StepCompletionFeature.State
+import ru.nobird.app.presentation.redux.reducer.StateReducer
+
+class StepCompletionReducer(private val stepRoute: StepRoute) : StateReducer<State, Message, Action> {
+    override fun reduce(state: State, message: Message): Pair<State, Set<Action>> =
+        when (message) {
+            is Message.ContinuePracticingClicked ->
+                if (!state.isPracticingLoading && state.currentStep.topic != null) {
```

Comment:
@vladkash 

When reducing the message `Message.ContinuePracticingClicked` if `state.currentStep.topic == null` means that we will do nothing. But the current topic id is only needed for several actions:
1. `Action.FetchNextRecommendedStep(state.currentStep)`
2. `Action.CheckTopicCompletionStatus(state.currentStep.topic)`

I think we still need have the ability to perform the remaining view actions when `state.currentStep.topic == null`:
1. `Action.ViewAction.NavigateTo.Back`
2. `Action.ViewAction.NavigateTo.HomeScreen`

What do you think?

Reply:
Yes, we have to do actions that could be performed without step topic